table/tables/bounded_tables.go

```diff
@@ -0,0 +1,260 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tables
+
+import (
+	"sync/atomic"
+
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/column"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/meta/autoid"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/util/types"
+)
+
+const (
+	initialRecordID int64 = 123
+	invalidRecordID int64 = 0
+)
+
+type boundedItem struct {
+	handle int64
+	data   []types.Datum
+}
+
+// BoundedTable implements table.Table interface.
+type BoundedTable struct {
+	ID          int64
+	Name        model.CIStr
+	Columns     []*column.Col
+	pkHandleCol *column.Col
+
+	recordPrefix kv.Key
+	alloc        autoid.Allocator
+	meta         *model.TableInfo
+
+	records  []boundedItem
+	capacity int32
+	cursor   int32
+}
+
+// BoundedTableFromMeta creates a Table instance from model.TableInfo.
+func BoundedTableFromMeta(alloc autoid.Allocator, tblInfo *model.TableInfo, capacity int32) (table.Table, error) {
+	columns := make([]*column.Col, 0, len(tblInfo.Columns))
+	var pkHandleColumn *column.Col
+	for _, colInfo := range tblInfo.Columns {
+		col := &column.Col{ColumnInfo: *colInfo}
+		columns = append(columns, col)
+		if col.IsPKHandleColumn(tblInfo) {
+			pkHandleColumn = col
+		}
+	}
+	t := newBoundedTable(tblInfo.ID, tblInfo.Name.O, columns, alloc, capacity)
+	t.pkHandleCol = pkHandleColumn
+	t.meta = tblInfo
+	return t, nil
```

Comment:
the function returns an error, but the implementation here always return a nil, so is it necessary?